packages/@lwc/babel-plugin-component/src/decorators/index.js

```diff
@@ -163,39 +139,133 @@ function generateInvalidDecoratorError(path) {
     }
 }
 
+function collectDecoratorPaths(bodyItems) {
+    return bodyItems.reduce((acc, bodyItem) => {
+        const decorators = bodyItem.get('decorators');
+        if (decorators && decorators.length) {
+            acc.push(...decorators);
+        }
+        return acc;
+    }, []);
+}
+
+function getDecoratorMetadata(decoratorPath) {
+    const expressionPath = decoratorPath.get('expression');
+
+    let name;
+    if (expressionPath.isIdentifier()) {
+        name = expressionPath.node.name;
+    } else if (expressionPath.isCallExpression()) {
+        name = expressionPath.node.callee.name;
+    } else {
+        throw generateInvalidDecoratorError(decoratorPath);
+    }
+
+    const propertyName = decoratorPath.parent.key.name;
+    const binding = decoratorPath.scope.getBinding(name);
+    const type = getDecoratorType(decoratorPath.parentPath);
+
+    return {
+        name,
+        propertyName,
+        binding,
```

Comment:
It is strange to expose the binding here. The binding can be retrieved from the `path` and `name` property. 
This can be removed from here.